test/suite/test_timestamp22.py

```diff
@@ -0,0 +1,298 @@
+
+        message += ' got ' + got
+
+        # If we're about to assert, show some extra info
+        if expected != got:
+            message += ': ERROR expected ' + expected
+            self.checkStderr()
+        self.pr(message)
+        self.assertEquals(expected, got)
+
+    # Create a predictable value based on the iteration number and timestamp.
+    def gen_value(self, iter, ts):
+        return str(iter) + '_' + str(ts) + '_' + 'x' * 1000
+
+    # Given a number representing an "approximate timestamp", generate a timestamp
+    # that is near that number, either plus or minus.
+    def gen_ts(self, approx_ts):
+        # a number between -10 and 10:
+        n = self.rand.rand32() % 21 - 10
+        ts = approx_ts + n
+        if ts <= 0:
+            ts = 1
+        return ts
+
+    # Asks whether we should do an illegal operation now. Return yes 5%.
+    def do_illegal(self):
+        return self.rand.rand32() % 20 == 0
+
+    def report(self, func, arg = None):
+        self.pr('DOING: ' + func + ('' if arg == None else '(' + arg + ')'))
+
+    # Insert a set of rows, each insert in its own transaction, with the
+    # given timestamps.
+    def updates(self, value, ds, do_prepare, commit_ts, durable_ts):
+        session = self.session
+        needs_rollback = False
+        prepare_config = None
+        commit_config = 'commit_timestamp=' + timestamp_str(commit_ts)
+
+        bad_commit = not do_prepare and self.do_illegal()
+        bad_prepare = False
+        # Occasionally put a durable timestamp on a commit without a prepare,
+        # that will be an error.
+        if do_prepare or bad_commit:
+            commit_config += ',durable_timestamp=' + timestamp_str(durable_ts)
+        cursor = session.open_cursor(self.uri)
+        prepare_ts = self.gen_ts(commit_ts)
+        prepare_config = 'prepare_timestamp=' + timestamp_str(prepare_ts)
+
+        # Predict whether the commit and optional prepare will fail.
+        if commit_ts < self.oldest_ts or commit_ts < self.stable_ts:
+            bad_commit = True
+        if do_prepare:
+            if commit_ts < prepare_ts:
+                bad_commit = True
+            if prepare_ts < self.oldest_ts:
+                bad_prepare = True
+            # If the prepare fails, the commit will as well.
+            if bad_prepare:
+                bad_commit = True
+        msg = 'inserts with commit config(' + commit_config + ')'
+
+        try:
+            for i in range(0, self.nrows):
+                needs_rollback = False
+                if self.do_illegal():
+                    # Illegal outside of transaction
+                    self.report('prepare_transaction', prepare_config)
+                    with self.expect(self.FAILURE, 'prepare outside of transaction'):
+                        session.prepare_transaction(prepare_config)
+
+                with self.expect(self.SUCCESS, 'begin_transaction'):
+                    session.begin_transaction()
```

Comment:
My earlier comment still stands - we could/should do some checking with `session.timestamp_transaction` instead of or in addition to setting it on the commit call. Is there anything we can do to add testing to changing the timestamp over the course of the transaction with that call? It could be follow-on work to this ticket (if so, perhaps add a comment about adding it, and where/when).